File: packages/cpp/src/scripts.cpp
#include "scripts.hpp"
#include <algorithm>
#include <cctype>

namespace lipilekhika {

static std::string to_lower(std::string s) {
    std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) {
        return std::tolower(c);
    });
    return s;
}

ScriptListEnum to_script_list_enum(Script s) {
    switch (s) {
        case Script::Devanagari:      return ScriptListEnum::Devanagari;
        case Script::Telugu:          return ScriptListEnum::Telugu;
        case Script::Tamil:           return ScriptListEnum::Tamil;
        case Script::TamilExtended:   return ScriptListEnum::TamilExtended;
        case Script::Bengali:         return ScriptListEnum::Bengali;
        case Script::Kannada:         return ScriptListEnum::Kannada;
        case Script::Gujarati:        return ScriptListEnum::Gujarati;
        case Script::Malayalam:       return ScriptListEnum::Malayalam;
        case Script::Odia:            return ScriptListEnum::Odia;
        case Script::Sinhala:         return ScriptListEnum::Sinhala;
        case Script::Normal:          return ScriptListEnum::Normal;
        case Script::Romanized:       return ScriptListEnum::Romanized;
        case Script::Gurumukhi:       return ScriptListEnum::Gurumukhi;
        case Script::Assamese:        return ScriptListEnum::Assamese;
        case Script::PurnaDevanagari: return ScriptListEnum::PurnaDevanagari;
        case Script::Brahmi:          return ScriptListEnum::Brahmi;
        case Script::Granth:          return ScriptListEnum::Granth;
        case Script::Modi:            return ScriptListEnum::Modi;
        case Script::Sharada:         return ScriptListEnum::Sharada;
        case Script::Siddham:         return ScriptListEnum::Siddham;
        case Script::English:         return ScriptListEnum::Normal;
        case Script::Sanskrit:        return ScriptListEnum::Devanagari;
        case Script::Hindi:           return ScriptListEnum::Devanagari;
        case Script::Marathi:         return ScriptListEnum::Devanagari;
        case Script::Nepali:          return ScriptListEnum::Devanagari;
        case Script::Punjabi:         return ScriptListEnum::Gurumukhi;
        case Script::Be:              return ScriptListEnum::Bengali;
        case Script::Rom:             return ScriptListEnum::Romanized;
        case Script::La:              return ScriptListEnum::Normal;
        case Script::Dev:             return ScriptListEnum::Devanagari;
        case Script::TamExt:          return ScriptListEnum::TamilExtended;
        case Script::Sin:             return ScriptListEnum::Sinhala;
        case Script::Tam:             return ScriptListEnum::Tamil;
        case Script::Ka:              return ScriptListEnum::Kannada;
        case Script::Kan:             return ScriptListEnum::Kannada;
        case Script::Si:              return ScriptListEnum::Sinhala;
        case Script::Eng:             return ScriptListEnum::Normal;
        case Script::Gur:             return ScriptListEnum::Gurumukhi;
        case Script::Nor:             return ScriptListEnum::Normal;
        case Script::Gu:              return ScriptListEnum::Gujarati;
        case Script::Mal:             return ScriptListEnum::Malayalam;
        case Script::Oriya:           return ScriptListEnum::Odia;
        case Script::En:              return ScriptListEnum::Normal;
        case Script::San:             return ScriptListEnum::Devanagari;
        case Script::Od:              return ScriptListEnum::Odia;
        case Script::Hin:             return ScriptListEnum::Devanagari;
        case Script::Ben:             return ScriptListEnum::Bengali;
        case Script::Te:              return ScriptListEnum::Telugu;
        case Script::Ne:              return ScriptListEnum::Devanagari;
        case Script::Sinh:             return ScriptListEnum::Sinhala;
        case Script::Nep:             return ScriptListEnum::Devanagari;
        case Script::Pun:             return ScriptListEnum::Gurumukhi;
        case Script::Hi:              return ScriptListEnum::Devanagari;
        case Script::Lat:             return ScriptListEnum::Normal;
        case Script::ValueAs:         return ScriptListEnum::Assamese;
        case Script::Norm:            return ScriptListEnum::Normal;
        case Script::Tel:             return ScriptListEnum::Telugu;
        case Script::Guj:             return ScriptListEnum::Gujarati;
        case Script::De:              return ScriptListEnum::Devanagari;
        case Script::TaExt:           return ScriptListEnum::TamilExtended;
        case Script::Or:              return ScriptListEnum::Odia;
        case Script::Mar:             return ScriptListEnum::Devanagari;
        case Script::Sa:              return ScriptListEnum::Devanagari;
    }
    return ScriptListEnum::Normal;
}

std::string to_string(ScriptListEnum s) {
    switch (s) {
        case ScriptListEnum::Devanagari:      return "Devanagari";
        case ScriptListEnum::Telugu:          return "Telugu";
        case ScriptListEnum::Tamil:           return "Tamil";
        case ScriptListEnum::TamilExtended:   return "Tamil-Extended";
        case ScriptListEnum::Bengali:         return "Bengali";
        case ScriptListEnum::Kannada:         return "Kannada";
        case ScriptListEnum::Gujarati:        return "Gujarati";
        case ScriptListEnum::Malayalam:       return "Malayalam";
        case ScriptListEnum::Odia:            return "Odia";
        case ScriptListEnum::Sinhala:         return "Sinhala";
        case ScriptListEnum::Normal:          return "Normal";
        case ScriptListEnum::Romanized:       return "Romanized";
        case ScriptListEnum::Gurumukhi:       return "Gurumukhi";
        case ScriptListEnum::Assamese:        return "Assamese";
        case ScriptListEnum::PurnaDevanagari: return "Purna-Devanagari";
        case ScriptListEnum::Brahmi:          return "Brahmi";
        case ScriptListEnum::Granth:          return "Granth";
        case ScriptListEnum::Modi:            return "Modi";
        case ScriptListEnum::Sharada:         return "Sharada";
        case ScriptListEnum::Siddham:         return "Siddham";
    }
    return "Normal";
}

std::string to_string(Script s) {
    switch (s) {
        case Script::Devanagari:      return "Devanagari";
        case Script::Telugu:          return "Telugu";
        case Script::Tamil:           return "Tamil";
        case Script::TamilExtended:   return "Tamil-Extended";
        case Script::Bengali:         return "Bengali";
        case Script::Kannada:         return "Kannada";
        case Script::Gujarati:        return "Gujarati";
        case Script::Malayalam:       return "Malayalam";
        case Script::Odia:            return "Odia";
        case Script::Sinhala:         return "Sinhala";
        case Script::Normal:          return "Normal";
        case Script::Romanized:       return "Romanized";
        case Script::Gurumukhi:       return "Gurumukhi";
        case Script::Assamese:        return "Assamese";
        case Script::PurnaDevanagari: return "Purna-Devanagari";
        case Script::Brahmi:          return "Brahmi";
        case Script::Granth:          return "Granth";
        case Script::Modi:            return "Modi";
        case Script::Sharada:         return "Sharada";
        case Script::Siddham:         return "Siddham";
        case Script::English:         return "English";
        case Script::Sanskrit:        return "Sanskrit";
        case Script::Hindi:           return "Hindi";
        case Script::Marathi:         return "Marathi";
        case Script::Nepali:          return "Nepali";
        case Script::Punjabi:         return "Punjabi";
        case Script::Be:              return "be";
        case Script::Rom:             return "rom";
        case Script::La:              return "la";
        case Script::Dev:             return "dev";
        case Script::TamExt:          return "tam-ext";
        case Script::Sin:             return "sin";
        case Script::Tam:             return "tam";
        case Script::Ka:              return "ka";
        case Script::Kan:             return "kan";
        case Script::Si:              return "si";
        case Script::Eng:             return "eng";
        case Script::Gur:             return "gur";
        case Script::Nor:             return "nor";
        case Script::Gu:              return "gu";
        case Script::Mal:             return "mal";
        case Script::Oriya:           return "oriya";
        case Script::En:              return "en";
        case Script::San:             return "san";
        case Script::Od:              return "od";
        case Script::Hin:             return "hin";
        case Script::Ben:             return "ben";
        case Script::Te:              return "te";
        case Script::Ne:              return "ne";
        case Script::Sinh:             return "sinh";
        case Script::Nep:             return "nep";
        case Script::Pun:             return "pun";
        case Script::Hi:              return "hi";
        case Script::Lat:             return "lat";
        case Script::ValueAs:         return "as";
        case Script::Norm:            return "norm";
        case Script::Tel:             return "tel";
        case Script::Guj:             return "guj";
        case Script::De:              return "de";
        case Script::TaExt:           return "ta-ext";
        case Script::Or:              return "or";
        case Script::Mar:             return "mar";
        case Script::Sa:              return "sa";
    }
    return "Normal";
}

std::optional<Script> script_from_string(const std::string& name) {
    std::string lower = to_lower(name);
    if (lower == "devanagari") return Script::Devanagari;
    if (lower == "telugu") return Script::Telugu;
    if (lower == "tamil") return Script::Tamil;
    if (lower == "tamil-extended" || lower == "tamilextended") return Script::TamilExtended;
    if (lower == "bengali") return Script::Bengali;
    if (lower == "kannada") return Script::Kannada;
    if (lower == "gujarati") return Script::Gujarati;
    if (lower == "malayalam") return Script::Malayalam;
    if (lower == "odia") return Script::Odia;
    if (lower == "sinhala") return Script::Sinhala;
    if (lower == "normal") return Script::Normal;
    if (lower == "romanized") return Script::Romanized;
    if (lower == "gurumukhi") return Script::Gurumukhi;
    if (lower == "assamese") return Script::Assamese;
    if (lower == "purna-devanagari" || lower == "purnadevanagari") return Script::PurnaDevanagari;
    if (lower == "brahmi") return Script::Brahmi;
    if (lower == "granth") return Script::Granth;
    if (lower == "modi") return Script::Modi;
    if (lower == "sharada") return Script::Sharada;
    if (lower == "siddham") return Script::Siddham;
    if (lower == "english") return Script::English;
    if (lower == "sanskrit") return Script::Sanskrit;
    if (lower == "hindi") return Script::Hindi;
    if (lower == "marathi") return Script::Marathi;
    if (lower == "nepali") return Script::Nepali;
    if (lower == "punjabi") return Script::Punjabi;
    if (lower == "be") return Script::Be;
    if (lower == "rom") return Script::Rom;
    if (lower == "la") return Script::La;
    if (lower == "dev") return Script::Dev;
    if (lower == "tam-ext") return Script::TamExt;
    if (lower == "sin") return Script::Sin;
    if (lower == "tam") return Script::Tam;
    if (lower == "ka") return Script::Ka;
    if (lower == "kan") return Script::Kan;
    if (lower == "si") return Script::Si;
    if (lower == "eng") return Script::Eng;
    if (lower == "gur") return Script::Gur;
    if (lower == "nor") return Script::Nor;
    if (lower == "gu") return Script::Gu;
    if (lower == "mal") return Script::Mal;
    if (lower == "oriya") return Script::Oriya;
    if (lower == "en") return Script::En;
    if (lower == "san") return Script::San;
    if (lower == "od") return Script::Od;
    if (lower == "hin") return Script::Hin;
    if (lower == "ben") return Script::Ben;
    if (lower == "te") return Script::Te;
    if (lower == "ne") return Script::Ne;
    if (lower == "sinh") return Script::Sinh;
    if (lower == "nep") return Script::Nep;
    if (lower == "pun") return Script::Pun;
    if (lower == "hi") return Script::Hi;
    if (lower == "lat") return Script::Lat;
    if (lower == "as") return Script::ValueAs;
    if (lower == "norm") return Script::Norm;
    if (lower == "tel") return Script::Tel;
    if (lower == "guj") return Script::Guj;
    if (lower == "de") return Script::De;
    if (lower == "ta-ext") return Script::TaExt;
    if (lower == "or") return Script::Or;
    if (lower == "mar") return Script::Mar;
    if (lower == "sa") return Script::Sa;

    return std::nullopt;
}

std::optional<ScriptListEnum> script_list_enum_from_string(const std::string& name) {
    std::string lower = to_lower(name);
    if (lower == "devanagari") return ScriptListEnum::Devanagari;
    if (lower == "telugu") return ScriptListEnum::Telugu;
    if (lower == "tamil") return ScriptListEnum::Tamil;
    if (lower == "tamil-extended" || lower == "tamilextended") return ScriptListEnum::TamilExtended;
    if (lower == "bengali") return ScriptListEnum::Bengali;
    if (lower == "kannada") return ScriptListEnum::Kannada;
    if (lower == "gujarati") return ScriptListEnum::Gujarati;
    if (lower == "malayalam") return ScriptListEnum::Malayalam;
    if (lower == "odia") return ScriptListEnum::Odia;
    if (lower == "sinhala") return ScriptListEnum::Sinhala;
    if (lower == "normal") return ScriptListEnum::Normal;
    if (lower == "romanized") return ScriptListEnum::Romanized;
    if (lower == "gurumukhi") return ScriptListEnum::Gurumukhi;
    if (lower == "assamese") return ScriptListEnum::Assamese;
    if (lower == "purna-devanagari" || lower == "purnadevanagari") return ScriptListEnum::PurnaDevanagari;
    if (lower == "brahmi") return ScriptListEnum::Brahmi;
    if (lower == "granth") return ScriptListEnum::Granth;
    if (lower == "modi") return ScriptListEnum::Modi;
    if (lower == "sharada") return ScriptListEnum::Sharada;
    if (lower == "siddham") return ScriptListEnum::Siddham;

    return std::nullopt;
}

} // namespace lipilekhika

File: packages/rust/src/scripts.rs
#[rustfmt::skip]
use strum::{AsRefStr, Display, EnumString};
/// The list of all supported scripts (internal resolved type)
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, AsRefStr, Display, EnumString)]
#[strum(ascii_case_insensitive)]
pub enum ScriptListEnum {
    #[strum(serialize = "Devanagari")]
    Devanagari,
    #[strum(serialize = "Telugu")]
    Telugu,
    #[strum(serialize = "Tamil")]
    Tamil,
    #[strum(serialize = "Tamil-Extended")]
    TamilExtended,
    #[strum(serialize = "Bengali")]
    Bengali,
    #[strum(serialize = "Kannada")]
    Kannada,
    #[strum(serialize = "Gujarati")]
    Gujarati,
    #[strum(serialize = "Malayalam")]
    Malayalam,
    #[strum(serialize = "Odia")]
    Odia,
    #[strum(serialize = "Sinhala")]
    Sinhala,
    #[strum(serialize = "Normal")]
    Normal,
    #[strum(serialize = "Romanized")]
    Romanized,
    #[strum(serialize = "Gurumukhi")]
    Gurumukhi,
    #[strum(serialize = "Assamese")]
    Assamese,
    #[strum(serialize = "Purna-Devanagari")]
    PurnaDevanagari,
    #[strum(serialize = "Brahmi")]
    Brahmi,
    #[strum(serialize = "Granth")]
    Granth,
    #[strum(serialize = "Modi")]
    Modi,
    #[strum(serialize = "Sharada")]
    Sharada,
    #[strum(serialize = "Siddham")]
    Siddham,
}
impl ScriptListEnum {
    /// Resolves a canonical `script_list.json` script id to the internal script enum.
    #[inline]
    pub const fn from_id(id: u8) -> Option<Self> {
        match id {
            1u8 => Some(Self::Devanagari),
            2u8 => Some(Self::Telugu),
            3u8 => Some(Self::Tamil),
            4u8 => Some(Self::TamilExtended),
            5u8 => Some(Self::Bengali),
            6u8 => Some(Self::Kannada),
            7u8 => Some(Self::Gujarati),
            8u8 => Some(Self::Malayalam),
            9u8 => Some(Self::Odia),
            10u8 => Some(Self::Sinhala),
            11u8 => Some(Self::Normal),
            12u8 => Some(Self::Romanized),
            13u8 => Some(Self::Gurumukhi),
            14u8 => Some(Self::Assamese),
            15u8 => Some(Self::PurnaDevanagari),
            16u8 => Some(Self::Brahmi),
            17u8 => Some(Self::Granth),
            18u8 => Some(Self::Modi),
            20u8 => Some(Self::Sharada),
            21u8 => Some(Self::Siddham),
            _ => None,
        }
    }
}
/// List of all supported scripts, languages and their aliases
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, AsRefStr, Display, EnumString)]
#[strum(ascii_case_insensitive)]
pub enum Script {
    #[strum(serialize = "Devanagari")]
    Devanagari,
    #[strum(serialize = "Telugu")]
    Telugu,
    #[strum(serialize = "Tamil")]
    Tamil,
    #[strum(serialize = "Tamil-Extended")]
    TamilExtended,
    #[strum(serialize = "Bengali")]
    Bengali,
    #[strum(serialize = "Kannada")]
    Kannada,
    #[strum(serialize = "Gujarati")]
    Gujarati,
    #[strum(serialize = "Malayalam")]
    Malayalam,
    #[strum(serialize = "Odia")]
    Odia,
    #[strum(serialize = "Sinhala")]
    Sinhala,
    #[strum(serialize = "Normal")]
    Normal,
    #[strum(serialize = "Romanized")]
    Romanized,
    #[strum(serialize = "Gurumukhi")]
    Gurumukhi,
    #[strum(serialize = "Assamese")]
    Assamese,
    #[strum(serialize = "Purna-Devanagari")]
    PurnaDevanagari,
    #[strum(serialize = "Brahmi")]
    Brahmi,
    #[strum(serialize = "Granth")]
    Granth,
    #[strum(serialize = "Modi")]
    Modi,
    #[strum(serialize = "Sharada")]
    Sharada,
    #[strum(serialize = "Siddham")]
    Siddham,
    #[strum(serialize = "English")]
    English,
    #[strum(serialize = "Sanskrit")]
    Sanskrit,
    #[strum(serialize = "Hindi")]
    Hindi,
    #[strum(serialize = "Marathi")]
    Marathi,
    #[strum(serialize = "Nepali")]
    Nepali,
    #[strum(serialize = "Punjabi")]
    Punjabi,
    #[strum(serialize = "de")]
    De,
    #[strum(serialize = "dev")]
    Dev,
    #[strum(serialize = "te")]
    Te,
    #[strum(serialize = "tel")]
    Tel,
    #[strum(serialize = "tam")]
    Tam,
    #[strum(serialize = "tam-ext")]
    TamExt,
    #[strum(serialize = "ta-ext")]
    TaExt,
    #[strum(serialize = "ben")]
    Ben,
    #[strum(serialize = "be")]
    Be,
    #[strum(serialize = "ka")]
    Ka,
    #[strum(serialize = "kan")]
    Kan,
    #[strum(serialize = "gu")]
    Gu,
    #[strum(serialize = "guj")]
    Guj,
    #[strum(serialize = "mal")]
    Mal,
    #[strum(serialize = "or")]
    Or,
    #[strum(serialize = "od")]
    Od,
    #[strum(serialize = "oriya")]
    Oriya,
    #[strum(serialize = "si")]
    Si,
    #[strum(serialize = "sinh")]
    Sinh,
    #[strum(serialize = "sin")]
    Sin,
    #[strum(serialize = "en")]
    En,
    #[strum(serialize = "eng")]
    Eng,
    #[strum(serialize = "la")]
    La,
    #[strum(serialize = "lat")]
    Lat,
    #[strum(serialize = "nor")]
    Nor,
    #[strum(serialize = "norm")]
    Norm,
    #[strum(serialize = "rom")]
    Rom,
    #[strum(serialize = "gur")]
    Gur,
    #[strum(serialize = "as")]
    As_,
    #[strum(serialize = "sa")]
    Sa,
    #[strum(serialize = "san")]
    San,
    #[strum(serialize = "hin")]
    Hin,
    #[strum(serialize = "hi")]
    Hi,
    #[strum(serialize = "mar")]
    Mar,
    #[strum(serialize = "ne")]
    Ne,
    #[strum(serialize = "nep")]
    Nep,
    #[strum(serialize = "pun")]
    Pun,
}
impl Script {
    /// Resolves a canonical `script_list.json` script id to the canonical script variant.
    #[inline]
    pub const fn from_id(id: u8) -> Option<Self> {
        match id {
            1u8 => Some(Self::Devanagari),
            2u8 => Some(Self::Telugu),
            3u8 => Some(Self::Tamil),
            4u8 => Some(Self::TamilExtended),
            5u8 => Some(Self::Bengali),
            6u8 => Some(Self::Kannada),
            7u8 => Some(Self::Gujarati),
            8u8 => Some(Self::Malayalam),
            9u8 => Some(Self::Odia),
            10u8 => Some(Self::Sinhala),
            11u8 => Some(Self::Normal),
            12u8 => Some(Self::Romanized),
            13u8 => Some(Self::Gurumukhi),
            14u8 => Some(Self::Assamese),
            15u8 => Some(Self::PurnaDevanagari),
            16u8 => Some(Self::Brahmi),
            17u8 => Some(Self::Granth),
            18u8 => Some(Self::Modi),
            20u8 => Some(Self::Sharada),
            21u8 => Some(Self::Siddham),
            _ => None,
        }
    }
}
impl From<Script> for ScriptListEnum {
    fn from(lang: Script) -> Self {
        match lang {
            Script::Devanagari => ScriptListEnum::Devanagari,
            Script::Telugu => ScriptListEnum::Telugu,
            Script::Tamil => ScriptListEnum::Tamil,
            Script::TamilExtended => ScriptListEnum::TamilExtended,
            Script::Bengali => ScriptListEnum::Bengali,
            Script::Kannada => ScriptListEnum::Kannada,
            Script::Gujarati => ScriptListEnum::Gujarati,
            Script::Malayalam => ScriptListEnum::Malayalam,
            Script::Odia => ScriptListEnum::Odia,
            Script::Sinhala => ScriptListEnum::Sinhala,
            Script::Normal => ScriptListEnum::Normal,
            Script::Romanized => ScriptListEnum::Romanized,
            Script::Gurumukhi => ScriptListEnum::Gurumukhi,
            Script::Assamese => ScriptListEnum::Assamese,
            Script::PurnaDevanagari => ScriptListEnum::PurnaDevanagari,
            Script::Brahmi => ScriptListEnum::Brahmi,
            Script::Granth => ScriptListEnum::Granth,
            Script::Modi => ScriptListEnum::Modi,
            Script::Sharada => ScriptListEnum::Sharada,
            Script::Siddham => ScriptListEnum::Siddham,
            Script::English => ScriptListEnum::Normal,
            Script::Sanskrit => ScriptListEnum::Devanagari,
            Script::Hindi => ScriptListEnum::Devanagari,
            Script::Marathi => ScriptListEnum::Devanagari,
            Script::Nepali => ScriptListEnum::Devanagari,
            Script::Punjabi => ScriptListEnum::Gurumukhi,
            Script::De => ScriptListEnum::Devanagari,
            Script::Dev => ScriptListEnum::Devanagari,
            Script::Te => ScriptListEnum::Telugu,
            Script::Tel => ScriptListEnum::Telugu,
            Script::Tam => ScriptListEnum::Tamil,
            Script::TamExt => ScriptListEnum::TamilExtended,
            Script::TaExt => ScriptListEnum::TamilExtended,
            Script::Ben => ScriptListEnum::Bengali,
            Script::Be => ScriptListEnum::Bengali,
            Script::Ka => ScriptListEnum::Kannada,
            Script::Kan => ScriptListEnum::Kannada,
            Script::Gu => ScriptListEnum::Gujarati,
            Script::Guj => ScriptListEnum::Gujarati,
            Script::Mal => ScriptListEnum::Malayalam,
            Script::Or => ScriptListEnum::Odia,
            Script::Od => ScriptListEnum::Odia,
            Script::Oriya => ScriptListEnum::Odia,
            Script::Si => ScriptListEnum::Sinhala,
            Script::Sinh => ScriptListEnum::Sinhala,
            Script::Sin => ScriptListEnum::Sinhala,
            Script::En => ScriptListEnum::Normal,
            Script::Eng => ScriptListEnum::Normal,
            Script::La => ScriptListEnum::Normal,
            Script::Lat => ScriptListEnum::Normal,
            Script::Nor => ScriptListEnum::Normal,
            Script::Norm => ScriptListEnum::Normal,
            Script::Rom => ScriptListEnum::Romanized,
            Script::Gur => ScriptListEnum::Gurumukhi,
            Script::As_ => ScriptListEnum::Assamese,
            Script::Sa => ScriptListEnum::Devanagari,
            Script::San => ScriptListEnum::Devanagari,
            Script::Hin => ScriptListEnum::Devanagari,
            Script::Hi => ScriptListEnum::Devanagari,
            Script::Mar => ScriptListEnum::Devanagari,
            Script::Ne => ScriptListEnum::Devanagari,
            Script::Nep => ScriptListEnum::Devanagari,
            Script::Pun => ScriptListEnum::Gurumukhi,
        }
    }
}
